test/groovy/PiperStageWrapperTest.groovy

```diff
@@ -115,5 +115,33 @@ class PiperStageWrapperTest extends BasePiperTest {
         assertThat(loggingRule.log, containsString('Config: ['))
         assertThat(loggingRule.log, containsString('testBranch'))
     }
-}
 
+    @Test
+    void testStageOldInterceptor() {
+        helper.registerAllowedMethod('fileExists', [String.class], { path ->
+            return (path == '.pipeline/extensions/test_old_extension.groovy')
+        })
+
+        helper.registerAllowedMethod('load', [String.class], {
+            return helper.loadScript('test/resources/stages/test_old_extension.groovy')
+        })
+        nullScript.commonPipelineEnvironment.gitBranch = 'testBranch'
+
+        def executed = false
+        stepRule.step.piperStageWrapper(
+            script: nullScript,
+            juStabUtils: utils,
+            ordinal: 10,
+            stageName: 'test_old_extension'
+        ) {
+            executed = true
+        }
+
+        assertThat(executed, is(true))
+        assertThat(loggingRule.log, containsString('[piperStageWrapper] Running project interceptor \'.pipeline/extensions/test_old_extension.groovy\' for test_old_extension.'))
+        assertThat(loggingRule.log, containsString('[Warning] The interface to implement extensions has changed.'))
+        assertThat(loggingRule.log, containsString('Stage Name: test_old_extension'))
+        assertThat(loggingRule.log, containsString('Config: ['))
+        assertThat(loggingRule.log, containsString('testBranch'))
```

Comment:
Minor: I think some of the assertions are not relevant for this test case

Reply:
The idea, though, was that whatever the testcase for the "new" interceptor API checks also needs to be checked for when the "old" interceptor API is used. You want to make sure that both code-paths work the same, but in addition the warning is in the log.